File: examples/java/regular_3.java
import CGAL.Kernel.Point_3;
import CGAL.Kernel.Weighted_point_3;
import CGAL.Triangulation_3.Regular_triangulation_3;
import java.util.Vector;


public class regular_3 {
  public static void main(String arg[]){

    // generate points on a 3D grid
    Vector<Weighted_point_3> P=new Vector<Weighted_point_3>();

    int number_of_points = 0;

    for (int z=0 ; z<5 ; z++)
      for (int y=0 ; y<5 ; y++)
        for (int x=0 ; x<5 ; x++) {
            Point_3 p = new Point_3(x, y, z);
            double w = (x+y-z*y*x)*2.0; // let's say this is the weight.
            P.add(new Weighted_point_3(p, w));
            ++number_of_points;
        }

    Regular_triangulation_3 T = new Regular_triangulation_3();

    // insert all points in a row (this is faster than one insert() at a time).
    T.insert (P.iterator());

    assert T.is_valid();
    assert T.dimension() == 3;

    System.out.println("Number of vertices : " + T.number_of_vertices());

    // removal of all vertices
    int count = 0;
    while (T.number_of_vertices() > 0) {
        T.remove (T.finite_vertices().next());
        ++count;
    }

    assert count == number_of_points;
  }
}

File: examples/python/regular_3.py
from __future__ import print_function
from CGAL.CGAL_Kernel import Point_3
from CGAL.CGAL_Kernel import Weighted_point_3
from CGAL.CGAL_Triangulation_3 import Regular_triangulation_3

# generate points on a 3D grid
P=[]

number_of_points = 0

for z in range(0,5):
  for y in range(0,5):
    for x in range(0,5):
        p = Point_3(x, y, z)
        w = (x+y-z*y*x)*2.0 # let's say this is the weight.
        P.append(Weighted_point_3(p, w))
        number_of_points+=1

T = Regular_triangulation_3()

# insert all points in a row (this is faster than one insert() at a time).
T.insert (P)

assert T.is_valid()
assert T.dimension() == 3

print("Number of vertices : ", T.number_of_vertices())

#removal of all vertices
count = 0
while T.number_of_vertices() > 0:
  T.remove (T.finite_vertices().next())
  count+=1


assert count == number_of_points
